grudge/dof_desc.py

```diff
@@ -58,6 +60,13 @@ class DTAG_VOLUME_ALL:  # noqa: N801
     """
 
 
+class DTAG_MODAL:  # noqa: N801
+    """
+    A tag denoting modal (rather than nodal)
+    degrees of freedom on cell volumes.
+    """
```

Comment:
This is not the right place for this. `DTAG` stands for "domain tag" (as you can tell from the surroundings), i.e. which geometry this is on. `QTAG` ("quadrature tag") describes how something is discretized. It's obviously possible to use modal discretizations on boundaries and volumes, so connecting these two concerns does not make sense.

`QTAG_MODAL` would be better, however "quadrature" is a misnomer. I'm open to renaming `QTAG`.

Reply:
I vote for `QTAG_MODAL` for now.

Reply:
To be fair, `DTAG_SCALAR` doesn't seem to make sense as a "domain tag" either. But I'll leave that for another day.

Reply:
> To be fair, `DTAG_SCALAR` doesn't seem to make sense as a "domain tag" either. But I'll leave that for another day.

Why doesn't it? The domain of the scalar is a set (any set) with a single element.

Reply:
Fixed in https://github.com/inducer/grudge/pull/77/commits/e9f6da0d85da1802a2b3f9b43ac32c5173ef65d9

Reply:
> Why doesn't it? The domain of the scalar is a set (any set) with a single element.

I think I was just incorrectly trying to pin too much "fem" type concepts to the tag when I initially looked at this. Disregard what I said. It's my misunderstanding.

Reply:
#79